src/category_theory/bicategory/functor.lean

```diff
@@ -212,10 +225,202 @@ def comp : oplax_functor B D :=
+(map₂_associator' : ∀ {a b c d : B} (f : a ⟶ b) (g : b ⟶ c) (h : c ⟶ d),
+  pseudofunctor.map₂_associator_aux obj (λ a b, map) (λ a b f g, map₂) (λ a b c, map_comp) f g h
+    . obviously)
+(map₂_left_unitor' : ∀ {a b : B} (f : a ⟶ b),
+  map₂ (λ_ f).hom = (map_comp (𝟙 a) f).hom ≫ ((map_id a).hom ▷ map f) ≫ (λ_ (map f)).hom
+    . obviously)
+(map₂_right_unitor' : ∀ {a b : B} (f : a ⟶ b),
+  map₂ (ρ_ f).hom = (map_comp f (𝟙 b)).hom ≫ (map f ◁ (map_id b).hom) ≫ (ρ_ (map f)).hom
+    . obviously)
+
+variables {B C}
+
+namespace pseudofunctor
+
+restate_axiom map₂_id'
+restate_axiom map₂_comp'
+restate_axiom map₂_whisker_left'
+restate_axiom map₂_whisker_right'
+restate_axiom map₂_associator'
+restate_axiom map₂_left_unitor'
+restate_axiom map₂_right_unitor'
+attribute [reassoc]
+  map₂_comp map₂_whisker_left map₂_whisker_right map₂_associator map₂_left_unitor map₂_right_unitor
+attribute [simp]
+  map₂_id map₂_comp map₂_whisker_left map₂_whisker_right
+  map₂_associator map₂_left_unitor map₂_right_unitor
+
+section
+variables (F : pseudofunctor B C) (G : pseudofunctor C D)
+
+open iso
+
+/-- The prelax functor between the underlying quivers. -/
+add_decl_doc pseudofunctor.to_prelax_functor
+
+instance has_coe_to_prelax_functor : has_coe (pseudofunctor B C) (prelax_functor B C) :=
+⟨to_prelax_functor⟩
+
+@[simp] lemma to_prelax_functor_eq_coe : F.to_prelax_functor = F := rfl
+@[simp] lemma to_prelax_functor_obj : (F : prelax_functor B C).obj = F.obj := rfl
+@[simp] lemma to_prelax_functor_map : (F : prelax_functor B C).map = F.map := rfl
+@[simp] lemma to_prelax_functor_map₂ : (F : prelax_functor B C).map₂ = F.map₂ := rfl
+
+/-- The oplax functor associated with a pseudofunctor. -/
+def to_oplax : oplax_functor B C :=
+{ map_id := λ a, (F.map_id a).hom,
+  map_comp := λ a b c f g, (F.map_comp f g).hom,
+  .. (F : prelax_functor B C) }
+
+instance has_coe_to_oplax : has_coe (pseudofunctor B C) (oplax_functor B C) := ⟨to_oplax⟩
+
+@[simp] lemma to_oplax_eq_coe : F.to_oplax = F := rfl
+@[simp] lemma to_oplax_obj : (F : oplax_functor B C).obj = F.obj := rfl
+@[simp] lemma to_oplax_map : (F : oplax_functor B C).map = F.map := rfl
+@[simp] lemma to_oplax_map₂ : (F : oplax_functor B C).map₂ = F.map₂ := rfl
+@[simp] lemma to_oplax_map_id (a : B) : (F : oplax_functor B C).map_id a = (F.map_id a).hom := rfl
+@[simp] lemma to_oplax_map_comp {a b c : B} (f : a ⟶ b) (g : b ⟶ c) :
+  (F : oplax_functor B C).map_comp f g = (F.map_comp f g).hom := rfl
+
+/-- Function on 1-morphisms as a functor. -/
+@[simps]
+def map_functor (a b : B) : (a ⟶ b) ⥤ (F.obj a ⟶ F.obj b) :=
+(F : oplax_functor B C).map_functor a b
+
+variables (B)
+
+/-- The identity pseudofunctor. -/
+@[simps]
+def id : pseudofunctor B B :=
+{ map_id := λ a, iso.refl (𝟙 a),
+  map_comp := λ a b c f g, iso.refl (f ≫ g),
+  .. prelax_functor.id B }
+
+instance : inhabited (pseudofunctor B B) := ⟨id B⟩
+
+variables {B}
+
+/-- Composition of pseudofunctors. -/
+@[simps]
+def comp : pseudofunctor B D :=
```

Comment:
Could I request not using `variables` every time it can be: I'd prefer that at major definitions the principal arguments are explicitly visible.